azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/ContainerService.java

```diff
@@ -0,0 +1,263 @@
+             * @return the stage representing configuration for the agent pool profile
+             */
+            CSAgentPoolProfile.DefinitionStages.Blank<WithCreate> defineContainerServiceAgentPoolProfile(String name);
+        }
+
+        /**
+         * The stage of the container service definition allowing the start of defining Linux specific settings.
+         */
+        interface WithLinuxProfile {
+            /**
+             * Begins the definition to specify Linux settings.
+             * @return the stage representing configuration of Linux specific settings.
+             */
+            WithLinuxProfileRootUsername withLinuxProfile();
+        }
+
+        /**
+         * The stage of the container service definition allowing to specific the Linux root username.
+         */
+        interface WithLinuxProfileRootUsername {
+            /**
+             * Begins the definition to specify Linux root username.
+             * @param rootUserName the root username
+             * @return the next stage of the definition
+             */
+            WithLinuxProfileSshKey withRootUsername(String rootUserName);
+        }
+
+        /**
+         * The stage of the container service definition allowing to specific the Linux ssh key.
+         */
+        interface WithLinuxProfileSshKey {
+            /**
+             * Begins the definition to specify Linux ssh key.
+             * @param sshKeyData the ssh key data.
+             * @return the next stage of the definition
+             */
+            DefineAgentPoolProfiles withSshKey(String sshKeyData);
+        }
+
+        /**
+         * The stage of the container service definition allowing to specific diagnostic settings.
+         */
+        interface WithDiagnosticsProfile {
+            /**
+             * Enable diagnostics.
+             * @return the create stage of the definition
+             */
+            WithCreate withDiagnostics();
+
+            /**
+             * Disable diagnostics.
+             * @return the create stage of the definition
+             */
+            WithCreate withoutDiagnostics();
+        }
+
+        /**
+         * The stage of the definition which contains all the minimum required inputs for
+         * the resource to be created (via {@link WithCreate#create()}), but also allows
+         * for any other optional settings to be specified.
+         */
+        interface WithCreate extends
+                Creatable<ContainerService>,
+                Resource.DefinitionWithTags<WithCreate>,
+                ContainerService.DefinitionStages.WithOrchestratorProfile,
+                ContainerService.DefinitionStages.WithDiagnosticsProfile {
+        }
+    }
+
+    /**
+     * The template for an update operation, containing all the settings that
+     * can be modified.
+     * <p>
+     * Call {@link Update#apply()} to apply the changes to the resource in Azure.

```

Comment:
For Update, to avoid `@link` references, I've been scrubbing things to remove that 'Call ...' sentence altogether.